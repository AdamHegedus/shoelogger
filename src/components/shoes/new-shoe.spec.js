import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import BootstrapVue from 'bootstrap-vue';
import NewShoe from '@/components/shoes/new-shoe';

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(BootstrapVue);

describe('new-shoe.vue', () => {
    describe('meta.status is false', () => {
        const shoesGetters = {
            getMeta: () => {
                return {
                    status: false,
                    message: 'Error message text.'
                };
            }
        };

        const shoesActions = {
            addShoe: jest.fn()
        };

        const brandsActions = {
            getBrands: jest.fn()
        };

        const typesActions = {
            getTypes: jest.fn()
        };

        const typesState = {
            types: [
                {
                    id: 0,
                    type: 'Foo'
                }
            ]
        };

        const brandsState = {
            brands: [
                {
                    id: 0,
                    brand: 'Foo'
                }
            ]
        };

        const store = new Vuex.Store({
            modules: {
                shoes: {
                    namespaced: true,
                    getters: shoesGetters,
                    actions: shoesActions
                },
                brands: {
                    namespaced: true,
                    state: brandsState,
                    actions: brandsActions
                },
                types: {
                    namespaced: true,
                    state: typesState,
                    actions: typesActions
                }
            }
        });

        it('should match snapshot when page loaded', () => {
            // ASSIGN
            const wrapper = shallowMount(NewShoe, { store, localVue });

            // ACT
            const actual = wrapper.element;

            // ASSERT
            expect(actual).toMatchSnapshot();
        });
    });

    describe('meta.status is true', () => {
        const shoesGetters = {
            getMeta: () => {
                return {
                    status: true,
                    message: null
                };
            }
        };

        const shoesActions = {
            addShoe: jest.fn()
        };

        const brandsActions = {
            getBrands: jest.fn()
        };

        const typesActions = {
            getTypes: jest.fn()
        };

        const typesState = {
            types: [
                {
                    id: 0,
                    type: 'Foo'
                }
            ]
        };

        const brandsState = {
            brands: [
                {
                    id: 0,
                    brand: 'Foo'
                }
            ]
        };

        const store = new Vuex.Store({
            modules: {
                shoes: {
                    namespaced: true,
                    getters: shoesGetters,
                    actions: shoesActions
                },
                brands: {
                    namespaced: true,
                    state: brandsState,
                    actions: brandsActions
                },
                types: {
                    namespaced: true,
                    state: typesState,
                    actions: typesActions
                }
            }
        });

        it('should match snapshot when page loaded', () => {
            // ASSIGN
            const wrapper = shallowMount(NewShoe, { store, localVue });

            // ACT
            const actual = wrapper.element;

            // ASSERT
            expect(actual).toMatchSnapshot();
        });
    });

    it('should match snapshot when text input changed', () => {
        // ASSIGN
        const shoesGetters = {
            getMeta: () => {
                return {
                    status: true,
                    message: null
                };
            }
        };

        const shoesActions = {
            addShoe: jest.fn()
        };

        const brandsActions = {
            getBrands: jest.fn()
        };

        const typesActions = {
            getTypes: jest.fn()
        };

        const typesState = {
            types: [
                {
                    id: 0,
                    type: 'Foo'
                }
            ]
        };

        const brandsState = {
            brands: [
                {
                    id: 0,
                    brand: 'Foo'
                }
            ]
        };

        const store = new Vuex.Store({
            modules: {
                shoes: {
                    namespaced: true,
                    getters: shoesGetters,
                    actions: shoesActions
                },
                brands: {
                    namespaced: true,
                    state: brandsState,
                    actions: brandsActions
                },
                types: {
                    namespaced: true,
                    state: typesState,
                    actions: typesActions
                }
            }
        });

        const wrapper = shallowMount(NewShoe, { store, localVue });
        const expected = 'Test';
        const input = 'Test';

        // ACT
        wrapper.find('input').setValue(input);

        // ASSERT
        const element = wrapper.element;
        const actual = wrapper.vm.$data.product;
        expect(element).toMatchSnapshot();
        expect(actual).toEqual(expected);
    });

    describe('getShoe', () => {
        it('should return the value of text input', () => {
            // ASSIGN
            const shoesGetters = {
                getMeta: () => {
                    return {
                        status: true,
                        message: null
                    };
                }
            };

            const shoesActions = {
                addShoe: jest.fn()
            };

            const brandsActions = {
                getBrands: jest.fn()
            };

            const typesActions = {
                getTypes: jest.fn()
            };

            const typesState = {
                types: [
                    {
                        id: 0,
                        type: 'Foo'
                    }
                ]
            };

            const brandsState = {
                brands: [
                    {
                        id: 0,
                        brand: 'Foo'
                    }
                ]
            };

            const store = new Vuex.Store({
                modules: {
                    shoes: {
                        namespaced: true,
                        getters: shoesGetters,
                        actions: shoesActions
                    },
                    brands: {
                        namespaced: true,
                        state: brandsState,
                        actions: brandsActions
                    },
                    types: {
                        namespaced: true,
                        state: typesState,
                        actions: typesActions
                    }
                }
            });

            const wrapper = shallowMount(NewShoe, { store, localVue });
            const input = 'Test';
            wrapper.vm.$data.selectedType = {
                id: 0,
                type: 'Foo'
            };
            wrapper.vm.$data.selectedBrand = {
                id: 0,
                type: 'Foo'
            };
            wrapper.find('#productName').setValue(input);

            // ACT
            const actual = wrapper.vm.getShoe;

            // ASSERT
            expect(actual).toMatchSnapshot();
        });
    });

    describe('brands', () => {
        it('should return the value from store', () => {
            // ASSIGN
            const shoesGetters = {
                getMeta: () => {
                    return {
                        status: true,
                        message: null
                    };
                }
            };

            const shoesActions = {
                addShoe: jest.fn()
            };

            const brandsActions = {
                getBrands: jest.fn()
            };

            const typesActions = {
                getTypes: jest.fn()
            };

            const typesState = {
                types: [
                    {
                        id: 0,
                        type: 'Foo'
                    }
                ]
            };

            const brandsState = {
                brands: [
                    {
                        id: 0,
                        brand: 'Foo'
                    }
                ]
            };

            const store = new Vuex.Store({
                modules: {
                    shoes: {
                        namespaced: true,
                        getters: shoesGetters,
                        actions: shoesActions
                    },
                    brands: {
                        namespaced: true,
                        state: brandsState,
                        actions: brandsActions
                    },
                    types: {
                        namespaced: true,
                        state: typesState,
                        actions: typesActions
                    }
                }
            });

            const wrapper = shallowMount(NewShoe, { store, localVue });

            // ACT
            const actual = wrapper.vm.brands;

            // ASSERT
            expect(actual).toMatchSnapshot();
        });
    });

    describe('types', () => {
        it('should return the value from store', () => {
            // ASSIGN
            const shoesGetters = {
                getMeta: () => {
                    return {
                        status: true,
                        message: null
                    };
                }
            };

            const shoesActions = {
                addShoe: jest.fn()
            };

            const brandsActions = {
                getBrands: jest.fn()
            };

            const typesActions = {
                getTypes: jest.fn()
            };

            const typesState = {
                types: [
                    {
                        id: 0,
                        type: 'Foo'
                    }
                ]
            };

            const brandsState = {
                brands: [
                    {
                        id: 0,
                        brand: 'Foo'
                    }
                ]
            };

            const store = new Vuex.Store({
                modules: {
                    shoes: {
                        namespaced: true,
                        getters: shoesGetters,
                        actions: shoesActions
                    },
                    brands: {
                        namespaced: true,
                        state: brandsState,
                        actions: brandsActions
                    },
                    types: {
                        namespaced: true,
                        state: typesState,
                        actions: typesActions
                    }
                }
            });

            const wrapper = shallowMount(NewShoe, { store, localVue });

            // ACT
            const actual = wrapper.vm.types;

            // ASSERT
            expect(actual).toMatchSnapshot();
        });
    });

    describe('addShoe', () => {
        it('should call action', () => {
            // ASSIGN
            const shoesGetters = {
                getMeta: () => {
                    return {
                        status: true,
                        message: null
                    };
                }
            };

            const shoesActions = {
                addShoe: jest.fn()
            };

            const brandsActions = {
                getBrands: jest.fn()
            };

            const typesActions = {
                getTypes: jest.fn()
            };

            const typesState = {
                types: [
                    {
                        id: 1,
                        type: 'Bar'
                    }
                ]
            };

            const brandsState = {
                brands: [
                    {
                        id: 0,
                        brand: 'Foo'
                    }
                ]
            };

            const store = new Vuex.Store({
                modules: {
                    shoes: {
                        namespaced: true,
                        getters: shoesGetters,
                        actions: shoesActions
                    },
                    brands: {
                        namespaced: true,
                        state: brandsState,
                        actions: brandsActions
                    },
                    types: {
                        namespaced: true,
                        state: typesState,
                        actions: typesActions
                    }
                }
            });

            const wrapper = shallowMount(NewShoe, { store, localVue });
            const input = {
                brandId: 0,
                typeId: 1,
                product: 'Test'
            };

            // ACT
            wrapper.vm.addShoe(input);

            // ASSERT
            const actual = shoesActions.addShoe.mock.calls;
            expect(actual).toMatchSnapshot();
        });
    });

    describe('selectBrand', () => {
        it('should set selectedBrand', () => {
            // ASSIGN
            const shoesGetters = {
                getMeta: () => {
                    return {
                        status: true,
                        message: null
                    };
                }
            };

            const shoesActions = {
                addShoe: jest.fn()
            };

            const brandsActions = {
                getBrands: jest.fn()
            };

            const typesActions = {
                getTypes: jest.fn()
            };

            const typesState = {
                types: [
                    {
                        id: 1,
                        type: 'Bar'
                    }
                ]
            };

            const brandsState = {
                brands: [
                    {
                        id: 0,
                        brand: 'Foo'
                    }
                ]
            };

            const store = new Vuex.Store({
                modules: {
                    shoes: {
                        namespaced: true,
                        getters: shoesGetters,
                        actions: shoesActions
                    },
                    brands: {
                        namespaced: true,
                        state: brandsState,
                        actions: brandsActions
                    },
                    types: {
                        namespaced: true,
                        state: typesState,
                        actions: typesActions
                    }
                }
            });

            const wrapper = shallowMount(NewShoe, { store, localVue });
            const input = {
                id: 0,
                brand: 'Foo'
            };

            // ACT
            wrapper.vm.selectBrand(input);

            // ASSERT
            const actual = wrapper.vm.selectedBrand;
            expect(actual).toMatchSnapshot();
        });
    });

    describe('selectType', () => {
        it('should set selectedType', () => {
            // ASSIGN
            const shoesGetters = {
                getMeta: () => {
                    return {
                        status: true,
                        message: null
                    };
                }
            };

            const shoesActions = {
                addShoe: jest.fn()
            };

            const brandsActions = {
                getBrands: jest.fn()
            };

            const typesActions = {
                getTypes: jest.fn()
            };

            const typesState = {
                types: [
                    {
                        id: 1,
                        type: 'Bar'
                    }
                ]
            };

            const brandsState = {
                brands: [
                    {
                        id: 0,
                        brand: 'Foo'
                    }
                ]
            };

            const store = new Vuex.Store({
                modules: {
                    shoes: {
                        namespaced: true,
                        getters: shoesGetters,
                        actions: shoesActions
                    },
                    brands: {
                        namespaced: true,
                        state: brandsState,
                        actions: brandsActions
                    },
                    types: {
                        namespaced: true,
                        state: typesState,
                        actions: typesActions
                    }
                }
            });

            const wrapper = shallowMount(NewShoe, { store, localVue });
            const input = {
                id: 1,
                brand: 'Bar'
            };

            // ACT
            wrapper.vm.selectType(input);

            // ASSERT
            const actual = wrapper.vm.selectedType;
            expect(actual).toMatchSnapshot();
        });
    });

    describe('getTypeDisplayValue', () => {
        it('should return correct value', () => {
            // ASSIGN
            const shoesGetters = {
                getMeta: () => {
                    return {
                        status: true,
                        message: null
                    };
                }
            };

            const shoesActions = {
                addShoe: jest.fn()
            };

            const brandsActions = {
                getBrands: jest.fn()
            };

            const typesActions = {
                getTypes: jest.fn()
            };

            const typesState = {
                types: [
                    {
                        id: 1,
                        type: 'Bar'
                    }
                ]
            };

            const brandsState = {
                brands: [
                    {
                        id: 0,
                        brand: 'Foo'
                    }
                ]
            };

            const store = new Vuex.Store({
                modules: {
                    shoes: {
                        namespaced: true,
                        getters: shoesGetters,
                        actions: shoesActions
                    },
                    brands: {
                        namespaced: true,
                        state: brandsState,
                        actions: brandsActions
                    },
                    types: {
                        namespaced: true,
                        state: typesState,
                        actions: typesActions
                    }
                }
            });

            const wrapper = shallowMount(NewShoe, { store, localVue });
            const input = {
                id: 1,
                type: 'Bar'
            };

            // ACT
            const actual = wrapper.vm.getTypeDisplayValue(input);

            // ASSERT
            expect(actual).toMatchSnapshot();
        });
    });

    describe('getBrandDisplayValue', () => {
        it('should return correct value', () => {
            // ASSIGN
            const shoesGetters = {
                getMeta: () => {
                    return {
                        status: true,
                        message: null
                    };
                }
            };

            const shoesActions = {
                addShoe: jest.fn()
            };

            const brandsActions = {
                getBrands: jest.fn()
            };

            const typesActions = {
                getTypes: jest.fn()
            };

            const typesState = {
                types: [
                    {
                        id: 1,
                        type: 'Bar'
                    }
                ]
            };

            const brandsState = {
                brands: [
                    {
                        id: 0,
                        brand: 'Foo'
                    }
                ]
            };

            const store = new Vuex.Store({
                modules: {
                    shoes: {
                        namespaced: true,
                        getters: shoesGetters,
                        actions: shoesActions
                    },
                    brands: {
                        namespaced: true,
                        state: brandsState,
                        actions: brandsActions
                    },
                    types: {
                        namespaced: true,
                        state: typesState,
                        actions: typesActions
                    }
                }
            });

            const wrapper = shallowMount(NewShoe, { store, localVue });
            const input = {
                id: 1,
                brand: 'Bar'
            };

            // ACT
            const actual = wrapper.vm.getBrandDisplayValue(input);

            // ASSERT
            expect(actual).toMatchSnapshot();
        });
    });
});
