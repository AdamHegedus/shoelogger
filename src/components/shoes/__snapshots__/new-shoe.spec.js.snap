// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new-shoe.vue addShoe should call action 1`] = `
Array [
  Array [
    Object {
      "commit": [Function],
      "dispatch": [Function],
      "getters": Object {
        "getMeta": Object {
          "message": null,
          "status": true,
        },
      },
      "rootGetters": Object {
        "shoes/getMeta": Object {
          "message": null,
          "status": true,
        },
      },
      "rootState": Object {
        "brands": Object {
          "brands": Array [
            Object {
              "brand": "Foo",
              "id": 0,
            },
          ],
        },
        "shoes": Object {},
        "types": Object {
          "types": Array [
            Object {
              "id": 1,
              "type": "Bar",
            },
          ],
        },
      },
      "state": Object {},
    },
    Object {
      "brandId": 0,
      "product": "Test",
      "typeId": 1,
    },
    undefined,
  ],
]
`;

exports[`new-shoe.vue brands should return the value from store 1`] = `
Array [
  Object {
    "brand": "Foo",
    "id": 0,
  },
]
`;

exports[`new-shoe.vue getShoe should return the value of text input 1`] = `
Object {
  "brandId": 0,
  "product": "Test",
  "typeId": 0,
}
`;

exports[`new-shoe.vue meta.status is false should match snapshot when page loaded 1`] = `
<div
  class="container"
>
  <div
    class="row"
  >
    <div
      class="col-sm col-lg-6"
    >
      <h2>
        Add New Shoe
      </h2>
       
      <form>
        <div
          class="form-group"
        >
          <label>
            Type
          </label>
           
          <b-dropdown-stub
            boundary="scrollParent"
            class="btn-block"
            id="type"
            no-flip="true"
            role="menu"
            size="lg"
            text="Select Type"
            toggleclass="btn btn-secondary btn-lg btn-block shadow"
            toggletext="Toggle Dropdown"
          >
            <b-dropdown-item-stub
              activeclass="active"
              class="[object Object]"
              event="click"
              exactactiveclass="active"
              routertag="a"
              target="_self"
            >
              
                            Foo
                        
            </b-dropdown-item-stub>
          </b-dropdown-stub>
        </div>
         
        <div
          class="form-group"
        >
          <label>
            Brand Name
          </label>
           
          <b-dropdown-stub
            boundary="scrollParent"
            class="btn-block"
            id="brandName"
            no-flip="true"
            role="menu"
            size="lg"
            text="Select Brand"
            toggleclass="btn btn-secondary btn-lg btn-block shadow"
            toggletext="Toggle Dropdown"
          >
            <b-dropdown-item-stub
              activeclass="active"
              class="[object Object]"
              event="click"
              exactactiveclass="active"
              routertag="a"
              target="_self"
            >
              
                            Foo
                        
            </b-dropdown-item-stub>
          </b-dropdown-stub>
        </div>
         
        <div
          class="form-group"
        >
          <label
            for="productName"
          >
            Product Name
          </label>
           
          <input
            class="form-control form-control-lg shadow border border-primary"
            id="productName"
            placeholder="Product Name"
            type="text"
          />
        </div>
         
        <button
          class="btn btn-primary btn-lg btn-block shadow"
          disabled="disabled"
          id="add"
          type="submit"
        >
          
                    Add
                
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`new-shoe.vue meta.status is true should match snapshot when page loaded 1`] = `
<div
  class="container"
>
  <div
    class="row"
  >
    <div
      class="col-sm col-lg-6"
    >
      <h2>
        Add New Shoe
      </h2>
       
      <form>
        <div
          class="form-group"
        >
          <label>
            Type
          </label>
           
          <b-dropdown-stub
            boundary="scrollParent"
            class="btn-block"
            id="type"
            no-flip="true"
            role="menu"
            size="lg"
            text="Select Type"
            toggleclass="btn btn-secondary btn-lg btn-block shadow"
            toggletext="Toggle Dropdown"
          >
            <b-dropdown-item-stub
              activeclass="active"
              class="[object Object]"
              event="click"
              exactactiveclass="active"
              routertag="a"
              target="_self"
            >
              
                            Foo
                        
            </b-dropdown-item-stub>
          </b-dropdown-stub>
        </div>
         
        <div
          class="form-group"
        >
          <label>
            Brand Name
          </label>
           
          <b-dropdown-stub
            boundary="scrollParent"
            class="btn-block"
            id="brandName"
            no-flip="true"
            role="menu"
            size="lg"
            text="Select Brand"
            toggleclass="btn btn-secondary btn-lg btn-block shadow"
            toggletext="Toggle Dropdown"
          >
            <b-dropdown-item-stub
              activeclass="active"
              class="[object Object]"
              event="click"
              exactactiveclass="active"
              routertag="a"
              target="_self"
            >
              
                            Foo
                        
            </b-dropdown-item-stub>
          </b-dropdown-stub>
        </div>
         
        <div
          class="form-group"
        >
          <label
            for="productName"
          >
            Product Name
          </label>
           
          <input
            class="form-control form-control-lg shadow border border-primary"
            id="productName"
            placeholder="Product Name"
            type="text"
          />
        </div>
         
        <button
          class="btn btn-primary btn-lg btn-block shadow"
          disabled="disabled"
          id="add"
          type="submit"
        >
          
                    Add
                
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`new-shoe.vue selectBrand should set selectedBrand 1`] = `
Object {
  "brand": "Foo",
  "id": 0,
}
`;

exports[`new-shoe.vue selectType should set selectedType 1`] = `
Object {
  "brand": "Bar",
  "id": 1,
}
`;

exports[`new-shoe.vue should match snapshot when text input changed 1`] = `
<div
  class="container"
>
  <div
    class="row"
  >
    <div
      class="col-sm col-lg-6"
    >
      <h2>
        Add New Shoe
      </h2>
       
      <form>
        <div
          class="form-group"
        >
          <label>
            Type
          </label>
           
          <b-dropdown-stub
            boundary="scrollParent"
            class="btn-block"
            id="type"
            no-flip="true"
            role="menu"
            size="lg"
            text="Select Type"
            toggleclass="btn btn-secondary btn-lg btn-block shadow"
            toggletext="Toggle Dropdown"
          >
            <b-dropdown-item-stub
              activeclass="active"
              class="[object Object]"
              event="click"
              exactactiveclass="active"
              routertag="a"
              target="_self"
            >
              
                            Foo
                        
            </b-dropdown-item-stub>
          </b-dropdown-stub>
        </div>
         
        <div
          class="form-group"
        >
          <label>
            Brand Name
          </label>
           
          <b-dropdown-stub
            boundary="scrollParent"
            class="btn-block"
            id="brandName"
            no-flip="true"
            role="menu"
            size="lg"
            text="Select Brand"
            toggleclass="btn btn-secondary btn-lg btn-block shadow"
            toggletext="Toggle Dropdown"
          >
            <b-dropdown-item-stub
              activeclass="active"
              class="[object Object]"
              event="click"
              exactactiveclass="active"
              routertag="a"
              target="_self"
            >
              
                            Foo
                        
            </b-dropdown-item-stub>
          </b-dropdown-stub>
        </div>
         
        <div
          class="form-group"
        >
          <label
            for="productName"
          >
            Product Name
          </label>
           
          <input
            class="form-control form-control-lg shadow border border-primary"
            id="productName"
            placeholder="Product Name"
            type="text"
          />
        </div>
         
        <button
          class="btn btn-primary btn-lg btn-block shadow"
          disabled="disabled"
          id="add"
          type="submit"
        >
          
                    Add
                
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`new-shoe.vue types should return the value from store 1`] = `
Array [
  Object {
    "id": 0,
    "type": "Foo",
  },
]
`;
